Extension/src/Debugger/extension.ts

```diff
@@ -79,38 +81,72 @@ export async function initialize(context: vscode.ExtensionContext): Promise<void
     await initializeSshTargets();
     const sshTargetsProvider: SshTargetsProvider = new SshTargetsProvider();
     disposables.push(vscode.window.registerTreeDataProvider('CppSshTargetsView', sshTargetsProvider));
-    disposables.push(vscode.commands.registerCommand(addSshTarget, addSshTargetImpl));
-    disposables.push(vscode.commands.registerCommand('C_Cpp.removeSshTarget', removeSshTargetImpl));
-    disposables.push(vscode.commands.registerCommand(refreshCppSshTargetsView, (node?: BaseNode) => sshTargetsProvider.refresh(node)));
+    disposables.push(vscode.commands.registerCommand(cmd_addSshTarget, () => enableSshTargetsViewAndRun(addSshTargetImpl)));
+    disposables.push(vscode.commands.registerCommand('C_Cpp.removeSshTarget', (node?: BaseNode) => enableSshTargetsViewAndRun(removeSshTargetImpl, node)));
+    disposables.push(vscode.commands.registerCommand(cmd_refreshCppSshTargetsView, (node?: BaseNode) => enableSshTargetsViewAndRun(sshTargetsProvider.refresh, node)));
     disposables.push(vscode.commands.registerCommand('C_Cpp.setActiveSshTarget', async (node: TargetLeafNode) => {
+        await enableSshTargetsView();
         await setActiveSshTarget(node.name);
-        await vscode.commands.executeCommand(refreshCppSshTargetsView);
+        await vscode.commands.executeCommand(cmd_refreshCppSshTargetsView);
     }));
-    disposables.push(vscode.commands.registerCommand('C_Cpp.selectSshTarget', selectSshTarget));
+    disposables.push(vscode.commands.registerCommand('C_Cpp.selectSshTarget', () => enableSshTargetsViewAndRun(selectSshTarget)));
     disposables.push(vscode.commands.registerCommand('C_Cpp.selectActiveSshTarget', async () => {
+        await enableSshTargetsView();
         const name: string | undefined = await selectSshTarget();
         if (name) {
             await setActiveSshTarget(name);
-            await vscode.commands.executeCommand(refreshCppSshTargetsView);
+            await vscode.commands.executeCommand(cmd_refreshCppSshTargetsView);
         }
     }));
-    disposables.push(vscode.commands.registerCommand('C_Cpp.activeSshTarget', getActiveSshTarget));
+    disposables.push(vscode.commands.registerCommand('C_Cpp.activeSshTarget', () => enableSshTargetsViewAndRun(getActiveSshTarget)));
     disposables.push(sshTargetsProvider);
-    for (const sshConfig of getSshConfigurationFiles()) {
-        fileWatchers.push(chokidar.watch(sshConfig, {ignoreInitial: true})
-            .on('add', () => vscode.commands.executeCommand(refreshCppSshTargetsView))
-            .on('change', () => vscode.commands.executeCommand(refreshCppSshTargetsView))
-            .on('unlink', () => vscode.commands.executeCommand(refreshCppSshTargetsView)));
+
+    // Decide if we should show the SSH Targets View.
+    sshTargetsViewSetting = (new CppSettings()).sshTargetsView;
+    if (sshTargetsViewSetting === 'Enabled') {
```

Comment:
If it's auto, do we want to show it if there's an active ssh target?  I didn't see whether we save the active target in PersistentWorkspaceState.

Reply:
Sure we can do that